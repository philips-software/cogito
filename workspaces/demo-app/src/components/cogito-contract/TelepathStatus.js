import React from 'react'
import { WithStore } from '@react-frontend-developer/react-redux-render-prop'
import {
  Row,
  Spacer
} from '@react-frontend-developer/react-layout-helpers'
import { Status } from 'components/styling'
import { Segment } from 'semantic-ui-react'

const TelepathStatus = ({ children }) => {
  return (
    <WithStore selector={state => ({
      telepathInProgress: state.appEvents.telepathInProgress
    })}
    >
      {
        ({ telepathInProgress }) => {
          if (telepathInProgress) {
            return (
              <Spacer margin='10px'>
                <Row>
                  <Segment>
                    <Status>{children}</Status>
                  </Segment>
                </Row>
              </Spacer>
            )
          } else {
            return null
          }
        }
      }
    </WithStore>
  )
}

export { TelepathStatus }
