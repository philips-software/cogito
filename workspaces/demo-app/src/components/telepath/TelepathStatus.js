import React from 'react'
import { WithStore } from '@react-frontend-developer/react-redux-render-prop'
import { StatusSegmentRow } from 'components/ui/layout'

const TelepathStatus = ({ children }) => {
  return (
    <WithStore selector={state => ({
      telepathInProgress: state.appEvents.telepathInProgress
    })}
    >
      {
        ({ telepathInProgress }) => {
          if (telepathInProgress) {
            return (
              <StatusSegmentRow>{children}</StatusSegmentRow>
            )
          } else {
            return null
          }
        }
      }
    </WithStore>
  )
}

export { TelepathStatus }
