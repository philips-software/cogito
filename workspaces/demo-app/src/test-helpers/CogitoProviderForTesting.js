class CogitoProviderForTesting {
  constructor ({ originalCogitoProvider, redirectProvider }) {
    this.cogitoProvider = originalCogitoProvider
    this.redirectProvider = redirectProvider
  }
  send (payload, callback) {
    if (payload.method === 'eth_accounts' ||
        payload.method === 'eth_sendTransaction') {
      this.redirectProvider.send(payload, callback)
    } else {
      this.cogitoProvider.send(payload, callback)
    }
  }

  sendAsync () {
    this.send(...arguments)
  }
}

export { CogitoProviderForTesting }
