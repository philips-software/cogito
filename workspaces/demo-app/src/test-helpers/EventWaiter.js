class EventWaiter {
  promise
  promiseResolve
  promiseReject
  expectedNumberOfEvents = 1
  intermediateValues = []

  constructor () {
    this.promise = new Promise((resolve, reject) => {
      this.promiseResolve = resolve
      this.promiseReject = reject
    })
  }

  reset () {
    this.promise = new Promise((resolve, reject) => {
      this.promiseResolve = resolve
      this.promiseReject = reject
    })
    this.expectedNumberOfEvents = 1
  }

  onValueChanged = value => {
    if (--this.expectedNumberOfEvents === 0) {
      this.promiseResolve(value)
      clearTimeout(this.timeout)
    } else {
      this.intermediateValues = [ ...this.intermediateValues, value ]
    }
  }

  expect (expectedNumberOfEvents) {
    this.expectedNumberOfEvents = expectedNumberOfEvents
  }

  wait () {
    this.timeout = setTimeout(() => {
      this.promiseReject(new Error('timedout'))
    }, 4000)
    return this.promise
  }
}

export { EventWaiter }
