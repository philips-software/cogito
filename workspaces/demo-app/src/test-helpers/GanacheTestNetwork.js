import Web3 from 'web3'
import ganache from 'ganache-cli'
import initContract from 'truffle-contract'

class GanacheTestNetwork {
  mnemonic = 'hair snack volcano shift tragic wrong wreck release vibrant gossip ugly debate'
  _web3

  constructor () {
    const provider = ganache.provider({
      mnemonic: this.mnemonic
    })
    this._web3 = new Web3(provider)
  }

  get web3 () {
    return this._web3
  }

  getAccounts = async () => {
    this.accounts = await this.web3.eth.getAccounts()
    return this.accounts
  }

  deploy = async (contractDefinition, { from }) => {
    const SimpleStorage = new this.web3.eth.Contract([contractDefinition])
    const deployedContract = await SimpleStorage.deploy({
      data: contractDefinition.bytecode
    }).send({
      from,
      gas: 1000000,
      gasPrice: '10000000000000'
    })
    const contract = initContract(contractDefinition)
    contract.setProvider(this.web3.currentProvider)

    return contract.at(deployedContract.options.address)
  }
}

export { GanacheTestNetwork }
