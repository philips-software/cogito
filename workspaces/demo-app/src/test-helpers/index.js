export { inRouter, renderInRouter } from './router'
export { EventWaiter } from './EventWaiter'
export { GanacheTestNetwork } from './GanacheTestNetwork'
