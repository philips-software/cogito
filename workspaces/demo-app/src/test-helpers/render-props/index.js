export {
  renderWithStore as render,
  rerenderWithStore as rerender
} from './renderWithStore'
export * from 'react-testing-library'
