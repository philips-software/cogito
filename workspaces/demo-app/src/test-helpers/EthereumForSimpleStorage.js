import { SimpleStorage } from '@cogitojs/demo-app-contracts'
import { CogitoEthereum } from '@cogitojs/cogito'
import { GanacheTestNetwork } from './GanacheTestNetwork'

class EthereumForSimpleStorage {
  ganacheTestNetwork
  context

  injectWeb3 = () => {
    window.web3 = this.ganacheTestNetwork.web3
    process.env.REACT_APP_USE_INJECTED_WEB3 = 'YES'
  }

  get address () {
    return this.ganacheTestNetwork.wallet.address
  }

  get telepathChannel () {
    return this.context.telepathChannel
  }

  get cogitoWeb3 () {
    return this.context.cogitoWeb3
  }

  get simpleStorageProxy () {
    return this.context.contractsProxies.SimpleStorage
  }

  constructor ({ appName }) {
    this.appName = appName
    this.ganacheTestNetwork = new GanacheTestNetwork()
    this.injectWeb3()
  }

  deploy = () => {
    return this.ganacheTestNetwork.deploy(SimpleStorage, { from: this.address })
  }

  setupCogitoContext = async contractBlob => {
    const cogitoEthereum = new CogitoEthereum([ contractBlob ])
    this.context = await cogitoEthereum.getContext({ appName: this.appName })

    this.ganacheTestNetwork.mockTelepathChannel(this.context.telepathChannel)
  }

  setup = async () => {
    const { deployedJSON } = await this.deploy()
    await this.setupCogitoContext(deployedJSON)
  }
}

export { EthereumForSimpleStorage }
