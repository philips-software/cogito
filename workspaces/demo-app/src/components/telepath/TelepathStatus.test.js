import React from 'react'
import { render } from 'test-helpers/render-props'
import { AppEventsActions } from 'app-events'
import { TelepathStatus } from './TelepathStatus'

jest.unmock('@react-frontend-developer/react-redux-render-prop')

describe('TelepathStatus', () => {
  const status = 'Status Message'

  it('shows status message when telepath opertion is in progress', () => {
    const { getByText, store: { dispatch } } = render(
      <TelepathStatus>{status}</TelepathStatus>
    )
    dispatch(AppEventsActions.telepathInProgress())
    expect(getByText(`${status}`)).toBeInTheDocument()
  })

  it('does not show anything if telepath operation is not in progress', () => {
    const { queryByText } = render(<TelepathStatus>{status}</TelepathStatus>)

    expect(queryByText(`${status}`)).toBeNull()
  })

  it('has correct styling when visble', () => {
    const { container, store: { dispatch } } = render(
      <TelepathStatus>{status}</TelepathStatus>
    )

    dispatch(AppEventsActions.telepathInProgress())

    expect(container).toMatchSnapshot()
  })

  it('renders nothing when not visible', () => {
    const { container } = render(
      <TelepathStatus>{status}</TelepathStatus>
    )

    expect(container).toMatchSnapshot()
  })
})
