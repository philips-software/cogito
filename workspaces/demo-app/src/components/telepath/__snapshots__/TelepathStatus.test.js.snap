// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelepathStatus has correct styling when visble 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.css-0,
[data-css-0] {
  font-size: 0.8em;
  color: #fd81c2;
}

<div>
  <div
    style="margin: 10px; padding: 0px; box-sizing: border-box;"
  >
    <div
      class="emotion-1 emotion-2"
    >
      <div
        class="ui segment"
      >
        <p
          class="css-0"
        >
          Status Message
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`TelepathStatus renders nothing when not visible 1`] = `<div />`;
